### ISSUE TITLE
Issue with free read-only queries (`eth_call`) using ethers.js

### ISSUE THREAD

**Issue by ed-marquez** [hashgraph/hedera-smart-contracts#576]
### Description

In the past year or so, we made read-only queries via the JSON-RPC free to align with what Eth developers expect (read-only queries used to be paid on Hedera - now they shouldn't be). 

The code below calls (`eth_call`) a view-only function (`getAmount`). The call fails if gas is set to zero. 
I had a similar example in a different repo that performed the view-only query successfully with zero gas a few months ago - that other example is now broken as well.

JS with Ethers:
```
	console.log(`- Call contract function (read only query) with eth_call...\n`);
	console.log(`- FAILS IF GAS IS ZERO - DOUBLE CHECK THIS LATER...\n`);

	gasLimit = 100000;
	const myContract = new ethers.Contract(contractAddress, abi, signer);
	const callTx = await myContract.getAmount(itemName, { gasLimit: gasLimit });
	const callResult = callTx.toString();

	console.log(`- Contract call complete ✅ | ${callResult} unit(s) of ${itemName} are available \n`);
```
Solidity function:
```
function getAmount(string memory _name) public view returns (uint) {
        return myInventory[_name];
    }
```

**Am I missing something? or has there been a change recently in how we do view-only queries?**

@logan

### Steps to reproduce

- Reference this example: https://github.com/ed-marquez/hedera-example-eth-call-estimategas-hip584/tree/main
- Open the repo with GitPod: [Click here to open on GitPod](https://gitpod.io/#https://github.com/ed-marquez/hedera-example-eth-call-estimategas-hip584)
- Enter your credentials in .env file
- In `index.js`, change the `gasLimit` in step 2 to `0`.
- Enter in the terminal the command: `node index.js`

### Hedera network

testnet, previewnet

### Version

"ethers": "^5.7.2", Hashio latest 

### Operating system

macOS

**Comment by Nana-EC** [comment]
Thanks @ed-marquez .
HIP 584 should be about complete as of Mirror Node 0.96 so we shouldn't be blocked by feature functionality - if anything just missed bugs.

We'll take a look and circle back

**Comment by ed-marquez** [comment]
Thanks @Nana-EC @ebadiere - this issue can be closed according to an internal thread in #ask-an-expert from Nov 9, 2023 at 11:12AM EST. Here are the details from the thread:

**Ed**
Queries via the JSON/RPC relay go to mirror -> `Free`
Read-only queries being `free` in Ethereum
**If a gas value is necessary, how are these read-only queries free?**

**Danno**
Free as in no ether/hbar is charged.
The gas limit is a limiter, but the ether per gas fee is set to zero.

**Ed**
By virtue of having to set a gas limit, do I still get charged 80% of whatever value I set? or does that also go to zero with the ether per gas being 0?

**Danno**
If its going to the mirror node you don't get charged.

**Ed**
Thanks all for the input on this. Good to know:
- that the call with zero gas was a bug that has been fixed (will update my other example code)
- ether per gas fee is set to zero
